src/modelfree/density/fit_density.py

```diff
@@ -0,0 +1,317 @@
+"""Fits density model to activations of victim's policy network."""
+
+import glob
+import json
+import logging
+import os
+import os.path as osp
+import pickle
+import re
+import tempfile
+
+import numpy as np
+import pandas as pd
+import ray
+import sacred
+from sacred.observers import FileStorageObserver
+from sklearn.decomposition import PCA
+from sklearn.mixture import GaussianMixture
+from sklearn.neighbors import KernelDensity
+
+from modelfree.common import utils
+
+fit_model_ex = sacred.Experiment('fit_density_model')
+logger = logging.getLogger('modelfree.density.fit_density')
+
+
+def gen_exp_name(model_class, model_kwargs):
+    """Generates experiment name from model class and parameters.
+
+    :param model_class: (type) the class, one of GaussianMixture, PCAPreDensity or KernelDensity.
+    :param model_kwargs: (dict) constructor arguments to the class.
+    :return A string succinctly encoding the class and parameters."""
+    if model_class == GaussianMixture:
+        n_components = model_kwargs.get('n_components', 1)
+        covariance_type = model_kwargs.get('covariance_type', 'full')
+        return f'gmm_{n_components}_components_{covariance_type}'
+    elif model_class == PCAPreDensity:
+        if model_kwargs['density_class'] == KernelDensity:
+            return 'pca_kde'
+        elif model_kwargs['density_class'] == 'pca_gmm':
```

Comment:
This seems like it'd only ever be called with 'density_class' = GaussianMixture, because this needs to be a callable for line 63, right?

Reply:
oops yes, fixed!